### DISCUSSION TITLE
After Update Pimcore 5.8.9 to 6.9.6 ... I get a 404 on /bundles/fosjsrouting/js/router.js

### DISCUSSION THREAD

**Discussion by phoebusryan** [pimcore/pimcore#17125] [Q&A / Support] [answered]
Hey there,
I have to update an old pimcore (5.8.9) which runs on a php 7.1 ubuntu server.

I found this tutorial:
https://pimcore.com/docs/pimcore/6.0/Development_Documentation/Installation_and_Upgrade/Upgrade_Notes/V5_to_V6.html

First I updated PHP to 7.2 then I replaces my composer.json with this one:
https://raw.githubusercontent.com/pimcore/skeleton/2.8/composer.json

I also created the database.yml from the tutorial and updated my web/app.php

... after that of course I did:
COMPOSER_MEMORY_LIMIT=-1 composer update

Now the pimcore is up and running again until I login. After the Login I get the error:
/bundles/fosjsrouting/js/router.js not found (404) ... and see an endless spinner in the middle of the page.

I googled a lot but I still have no idea what's wrong here. Can you please help me? :(

**Comment by markus-moser** [comment]
Please execute `bin/console asset:install --symlink` and see if this helps...

**Reply by phoebusryan** [reply]
Thanks for your response. I just tried that and got the error 'target directory "../public" does not exist.
![pimcore_error](https://github.com/pimcore/pimcore/assets/4052863/e2a0ba48-edff-446b-8c58-4ee15f18407b)

**Reply by phoebusryan** [reply]
Ah, I just tried this:
https://stackoverflow.com/a/63577749

And then again your command. I think it's fixed now because I get now two different JS-Errors after Login:
```
The route "pimcore_admin_document_document_doctypesget" does not exist.
The route "pimcore_admin_notification_findlastunread" does not exist.
```

In the PIM Logs, I found the error "Table 'pimcore.lock_keys' doesn't exist". I don't know why. Than I checked via Mysql the table "pimcore_migrations". There are only very old records. I think the migration script didn't run. Hm...

I retried "composer update" and a errormessage appeared:
"Symfony\Component\Process\Process::__construct() must be of the type array, string given"

I think this is because my composer.json gets a newer symfony-version. So I tried ...:
```
composer require --no-update symfony/symfony:^4.0
composer require --no-update symfony/process:^4.0

COMPOSER_MEMORY_LIMIT=-1 composer update
```

But it did not help. The errormessage is still the same.

**Reply by markus-moser** [reply]
You could try to do a `bin/console doctrine:migrations:migrate` to execute missing migrations. Not sure but maybe the migration command was a different one for Pimcore 6.

**Reply by markus-moser** [reply, marked as answer]
The right command for Pimcore 6 should be `bin/console pimcore:migrations:migrate -s pimcore_core -n`

**Reply by phoebusryan** [reply]
Oh that worked! :) Thanks man! Now i will do a step further to pimcore 10.
